scripts/pre_commit_linter.py

```diff
@@ -881,8 +883,91 @@ def _check_html_directive_name(all_files):
+                        'MemberExpression')) and (
+                            expression['callee']['property']['name'] == (
+                                'directive'))):
+                continue
+            arguments = expression['arguments']
+            directive_name = str(arguments[0]['value']) if (
+                arguments[0]['type'] == 'Literal') else (
+                    "No such directive exists.")
+            arguments = arguments[1:]
+            for argument in arguments:
+                if argument['type'] != 'ArrayExpression':
+                    continue
+                elements = argument['elements']
+                for element in elements:
+                    if element['type'] != 'FunctionExpression':
+                        continue
+                    body = element['body']
+                    if body['type'] != 'BlockStatement':
+                        continue
+                    body_elements = body['body']
+                    for body_element in body_elements:
+                        if not (body_element['type'] == 'ReturnStatement' and (
+                                body_element['argument']['type'] == 'ObjectExpression')):
+                            continue
+                        return_node_properties = body_element['argument']['properties']
+                        for return_node_property in return_node_properties:
+                            if return_node_property['key']['type'] == 'Identifier' and (
+                                return_node_property['key']['name'] == 'scope'):
+                                scope_value = return_node_property['value']
+                                if scope_value['type'] == 'ObjectExpression':
+                                    directive_has_isolated_scope = True
+                                elif scope_value['type'] == 'Literal' and scope_value['value']:
+                                    failed = True
+                                    print 'Please ensure that %s directive in %s file does not have scope set to true.' %(directive_name, filename)
+            if not directive_has_isolated_scope:
+                failed = True
+                print 'Please ensure that %s directive in %s file has a scope: {}.' %(directive_name, filename)                                                
+
+    if failed:
+        summary_message = '%s   Directive scope check failed' % (
+            _MESSAGE_TYPE_FAILED)
+        print summary_message
+        summary_messages.append(summary_message)
+    else:
+        summary_message = '%s  Directive scope check passed' % (
+            _MESSAGE_TYPE_SUCCESS)
+        print summary_message
+        summary_messages.append(summary_message)
+
+    print ''
+    print '----------------------------------------'
+    print ''
+
+    return summary_messages
+
+
 def main():
     all_files = _get_all_files()
+    # directive_scope_messages = _check_directive_scope(all_files)
```

Comment:
Don't leave commented-out code here. If you need to change something add a TODO instead and make sure there's an issue filed for it.

Reply:
Done.